src/middleware/inject_params.rs

```diff
@@ -6,74 +6,93 @@ use jsonrpsee::{
 use std::sync::Arc;
 
 use super::{Middleware, NextFn};
-use crate::{
-    api::{Api, ValueHandle},
-    config::MethodParam,
-    middleware::call::CallRequest,
-};
+use crate::{api::Api, config::MethodParam, middleware::call::CallRequest};
 
 pub enum InjectType {
     BlockHashAt(usize),
     BlockNumberAt(usize),
+    BlockTagAt(usize),
 }
 
 pub struct InjectParamsMiddleware {
-    head: ValueHandle<(JsonValue, u64)>,
+    api: Arc<Api>,
```

Comment:
why?

Reply:
so I have access to api not just head

Reply:
but it looks like you only access head and finalized_head